src/webview/MessageList.js

```diff
@@ -125,21 +128,22 @@ function useMessageListProps(props: OuterProps): Props {
   const globalSettings = useGlobalSelector(getGlobalSettings);
   const debug = useGlobalSelector(getDebug);
 
-  const doNotMarkMessagesAsRead =
-    !marksMessagesAsRead(props.narrow)
-    || (() => {
-      switch (globalSettings.markMessagesReadOnScroll) {
-        case 'always':
-          return false;
-        case 'never':
-          return true;
-        case 'conversation-views-only':
-          return !isConversationNarrow(props.narrow);
-        default:
-          ensureUnreachable(globalSettings.markMessagesReadOnScroll);
-          return false;
-      }
-    })();
+  const [doNotMarkMessagesAsRead, setDoNotMarkMessagesAsRead] = useState(() => {
```

Comment:
Yeah, good thought.

In fact, we already weren't set up to handle `doNotMarkMessagesAsRead` changing. It's used in generating the initial HTML, where it initializes a variable of the same name that's consulted in `js.js` — and we didn't have an inbound event to update that if it changes. This is actually a bug in this version of the PR, because it means that the `setDoNotMarkMessagesAsRead(true)` in the action-sheet action fails to stop the green unread-indicators from getting erased. The next revision will fix that.

I think it may not be too complicated to write this so that it works straightforwardly if the desired initial value did somehow change. So I may just do that.

> In fact, `narrow` does change in a few cases:

(Happily, though, neither of those cases changes from one type of narrow to another; so `isConversationNarrow(narrow)` remains constant, and the initial value of `doNotMarkMessagesAsRead` isn't affected.)

Reply:
> (Happily, though, neither of those cases changes from one type of narrow to another; so isConversationNarrow(narrow) remains constant, and the initial value of doNotMarkMessagesAsRead isn't affected.)

True! I had reasoned through that too, but thought it might be good to put in a code comment, and also it's worth knowing whether we might one day add a case that changes the narrow type.